src/common/utils.cpp

```diff
@@ -53,13 +65,14 @@ std::string getDBusServiceName(sdbusplus::bus::bus& bus,
     // In OpenBMC, the object path will be hosted by a single service
     // i.e more than one service cannot host the same object path.
     // Note that for legacy reasons, phosphor-state-manager registers two
-    // service names, xyz.openbmc_project.State.Host and
-    // xyz.openbmc_project.State.Host0. This was to support multi-host
+    // service names, (xyz.openbmc_project.State.Host and
+    // xyz.openbmc_project.State.Host0) or (xyz.openbmc_project.State.Chassis
+    // and xyz.openbmc_project.State.Chassis0). This was to support multi-host
     // designs but also support legacy users. This is the one exception
     // to the "more than one service" rule
     if ((servicesName.size() > 1) &&
-        (servicesName[0].first.find("xyz.openbmc_project.State.Host") ==
-         std::string::npos))
+        isNotStateHostService(servicesName[0].first) &&
```

Comment:
make it an inline function

I would do it differently by using [any_of](https://en.cppreference.com/w/cpp/algorithm/ranges/all_any_none_of) with **lambda** so that it can be easily extendable and maintainable.

For example

```
auto isContainExceptionalServices = [] (const auto& serviceInfo) {
     std::set<std::string> exceptionsServices = {"xyz.openbmc_project.State.Host",
                                                 "xyz.openbmc_project.State.Chassis"};

     return exceptionsServices.contains(serviceInfo.first);
};

if ( (servicesName.size() > 1) &&
     (! std::ranges::any_of(servicesName, isContainExceptionalServices)))
{
   // error
}
```

Note, this is not tested code just an example so please feel free to edit.

Reply:
Sure